File: public/dsa/linked_lists/print_reversed_linked_list/approach/print_reversed_linked_list.cpp

#include <bits/stdc++.h>
using namespace std;

// defs-start
class ListNode {
   public:
    int data;
    ListNode* next;

    ListNode(int data) {
        this->data = data;
        this->next = NULL;
    }
};
// defs-end

void printLinkedListReverse(ListNode* head) {
    // add your logic here
    if (head) {
        printLinkedListReverse(head->next);
        cout << head->data << " ";
    }
}

// tests-start
int main() {
    ListNode* head = new ListNode(1);
    head->next = new ListNode(3);
    head->next->next = new ListNode(4);
    head->next->next->next = new ListNode(7);

    printLinkedListReverse(head);

    return 0;
}
// tests-end

/*output
7 4 3 1
*/

File: public/dsa/linked_lists/print_reversed_linked_list/approach/print_reversed_linked_list.py
# defs-start
class ListNode:
    def __init__(self, data=0, next=None):
        self.data = data
        self.next = next
# defs-end


class Solution:
    def printLinkedListReverse(self, head: ListNode) -> None:
        if head:
            self.printLinkedListReverse(head.next)
            print(head.data, end=" ")


# tests-start
head = ListNode(1, ListNode(3, ListNode(4, ListNode(7))))
sol = Solution()
sol.printLinkedListReverse(head)
# tests-end

'''output
7 4 3 1
'''
